### PR TITLE
T-104 / Product controller tests & changes

### PR DESCRIPTION
**Pull request by fiorellamansilla**
- Changes on Product Controller endpoints & ApiResponse class
- Implemented tests fo the Product Controller endpoints

### CODE REVIEW

File: src/main/java/com/kuna_backend/controllers/ProductController.java [hunk 1]
```diff
@@ -78,10 +77,10 @@ public ResponseEntity<ApiResponse> createProductVariationForProduct(@PathVariabl
     // TODO: TEST THIS POST ENDPOINT TO VERIFY THAT ONLY ATTRIBUTES FROM THE PRODUCT HAVE BEEN UPDATED.
     // UPDATE a specific Product by ID - Endpoint
     @PostMapping(path = "/update/{productId}")
-    public ResponseEntity<ApiResponse> updateProduct(@PathVariable("productId") Integer productId, @RequestBody ProductDto productDto) {
+    public ResponseEntity<ApiResponse> updateProductOnly(@PathVariable("productId") Integer productId, @RequestBody ProductDto productDto) {
```
Comment (diego-mansilla) [inline src/main/java/com/kuna_backend/controllers/ProductController.java:80]: What is updateProductonly? One product?

File: src/main/java/com/kuna_backend/services/ProductService.java [hunk 2]
```diff
@@ -67,7 +67,35 @@ public List<ProductDto> listProducts(Integer pageNumber, Integer pageSize) {
...
+    public Product updateProduct(Integer productId, ProductDto updatedProductDto, Category category) {
+
+        // Retrieve the specific Product
+        Product product = getProductById(productId);
+
+        product.setName(updatedProductDto.getName());
+        product.setPrice(updatedProductDto.getPrice());
+        product.setDescription(updatedProductDto.getDescription());
+        product.setImageUrl(updatedProductDto.getImageUrl());
+        product.setCreatedAt(new Timestamp(System.currentTimeMillis()).toLocalDateTime());
```
Comment (diego-mansilla) [inline src/main/java/com/kuna_backend/services/ProductService.java:80]: If this is update method why are you changing createdAt

File: src/main/java/com/kuna_backend/controllers/ProductController.java (hunk 1 above)
Comment (fiorellamansilla) [reply src/main/java/com/kuna_backend/controllers/ProductController.java:80]: It means that only the Product attributes are modified with this update endpoint. Should I modify the name of the endpoint?

File: src/main/java/com/kuna_backend/services/ProductService.java (hunk 2 above)
Comment (fiorellamansilla) [reply src/main/java/com/kuna_backend/services/ProductService.java:80]: I've just made the corrections to this method.

**Review by diego-mansilla — APPROVED** [review]

**Pull request merged**